convai_evaluation.py

```diff
@@ -57,12 +58,14 @@ def __init__(self, opt, shared=None):
             self.logger.info("Get pretrained model and tokenizer")
             if args.model_checkpoint == "":
                 args.model_checkpoint = download_pretrained_model()
-
-            self.tokenizer = OpenAIGPTTokenizer.from_pretrained(args.model_checkpoint)
-            if self.args.eval_type == "hits@1":
-                self.model_checkpoint = OpenAIGPTDoubleHeadsModel.from_pretrained(args.model_checkpoint)
+            if 'gpt2' in args.model_checkpoint:
```

Comment:
I think if someone is passing in a path to a model checkpoint, they're not going to have 'gpt2' in that path name by default. Maybe we can separate model_checkpoint into model_name and model_checkpoint, where model_name can be openai-gpt and gpt2 and model_checkpoint will be "" or a path to a run.

Reply:
I'm with you 100%, but `train.py` makes the same assumption and I want to keep the PR small. At the very least, `train.py` should generate model checkpoints with 'gpt2' in the path if the next step expects that.

Reply:
Ok. We can add a line after line 249 in `train.py` to set `log_dir = os.path.join(log_dir, args.model_checkpoint' perhaps.